File: 349. Intersection of Two Arrays/_349.py
class Solution:
    def intersection(self, nums1, nums2):
        mydict = {}
        res = list()

        for i in nums1:
            mydict[i] = 1
        
        for i in nums2:
            v = mydict.get(i, 0)
            if v != 0:
                res.append(i)
                mydict[i] = 0

        return res

    # 去重要想到set
    def intersection1(self, nums1, nums2):
        return list(set(nums1) & set(nums2))

File: 349. Intersection of Two Arrays/349.go
package leetcode

func intersection(nums1 []int, nums2 []int) []int {
	mydict := make(map[int]int, 0)
	res := make([]int, 0)
	for _, v := range nums1{
		if _, ok := mydict[v]; !ok{
			mydict[v] = 1
		}
	}
	// 上面这样写更好
	// for i:=0;i<len(nums1);i++{
    //     mydict[nums1[i]]=1
    // }

	for _, v := range nums2{
		if dictv, ok := mydict[v]; ok&&dictv == 1{
			res = append(res, v)
			mydict[v] = 0
		}
	}
	return res
}